### ISSUE TITLE
Monitoring - the app cluster-monitoring is stuck on installing state

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#20166]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- run Rancher: v2.3.0-alpha1
- add a REK ec2 cluster (3 etcd, 2 control plane, 3 workers)
- enable the cluster monitoring 

**Result:**
- the app `cluster-monitoring` is stuck on `installing` state but all workloads are `active` and the cluster monitoring is functional. 
<img width="1320" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/57560148-d6778d00-7339-11e9-966b-762d4f53efa2.png">

**Other details that may be helpful:**
- turn the cluster monitoring off and on will get the app re-deployed and eventually being `active`  
- It does not happen all the time, for example, it does not happen on another EC2 cluster  (3 etcd, 1 control plane, 3 workers) on the same Rancher server 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.3.0-alpha1
- Installation option (single install/HA): 
single node 

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.6", GitCommit:"ab91afd7062d4240e95e51ac00a18bd58fddd365", GitTreeState:"clean", BuildDate:"2019-02-26T12:59:46Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
 18.9.6
```

**Comment by loganhz** [comment]
Does refresh help?

**Comment by loganhz** [comment]
not sure if it is related to https://github.com/rancher/rancher/issues/20202#issuecomment-492062949

Can you test again 2.2.3, please? The fix is not in v2.3.0-alpha1

**Comment by jiaqiluo** [comment]
The bug fix is validated on Rancher: master beffccd

Steps:
-  enable and disable the cluster monitoring on several different clusters

Result:
- did not see the state of the `cluster-monitoring` app being stuck on `installing`  anymore
- occasionally saw the following situation, but it goes normal after refreshing the page  (the same observation result on v2.2.4-rc1 )
<img width="1489" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/57726234-d5997080-7643-11e9-8b28-f71005dfd2b5.png">
